src/facade/reply_builder.cc

```diff
@@ -169,6 +169,7 @@ MCReplyBuilder::MCReplyBuilder(::io::Sink* sink) : SinkReplyBuilder(sink), norep
 }
 
 void MCReplyBuilder::SendSimpleString(std::string_view str) {
+  has_replied_ = true;
```

Comment:
How about MCReplyBuilder::SendClientError? it does not use SendSimpleString

Reply:
uses `SinkReplyBuilder::Send` :)

Reply:
but SendSimpleString also uses Send, so why did you need to set has_replied_= true here? 
Is is related to line 174 return? why do you set has_replied_ is noreply_ is true?

Reply:
In this case we say explicitly that we should not reply. Instead of setting has_replied_ to true, I believe you should  not set ExpectReply in the ReplyGuard in this case